python/src/array.cpp

```diff
@@ -294,6 +294,26 @@ void init_array(nb::module_& m) {
             Returns:
                 array: The array with type ``dtype``.
           )pbdoc")
+      .def(
+          "__array_namespace__",
+          [](const array& a, const std::optional<std::string>& api_version) {
+            if (api_version) {
+              throw std::invalid_argument(
+                  "Explicitly specifying api_version is not yet implemented.");
+            }
+            return nb::module_::import_("mlx.core");
+          },
+          "api_version"_a = nb::none(),
+          R"pbdoc(
+            Used to apply updates at the given indices.
```

Comment:
Nit: doc string? Took me a sec to figure out this is how einops knows to call `mx.permute_dims` etc

Reply:
Ah that's a copy typo, good catch.